rllib/algorithms/alpha_zero/alpha_zero.py

```diff
@@ -169,6 +169,7 @@ def training(
         vf_share_layers: Optional[bool] = None,
         mcts_config: Optional[dict] = None,
         ranked_rewards: Optional[dict] = None,
+        num_steps_sampled_before_learning_starts: Optional[int] = None,
```

Comment:
Should this be part of AlgorithmConfig instead of each specific algorithm? We need to put it in a place that is generally accessible by all other configs.

Reply:
The thing is that the on-policy algorithms don't use it and I can't think of a reason right now why they should ever need this parameter. What do you say?

Reply:
yeah but copying this code over and over also seems a bit of waste. We had replay buffer parameter before but it wasn't used by any of the on-policy algorithms. It makes more sense for it to be part of the base class.

Reply:
I see, but this is the case for most of the parameters we use.
For example "lr_schedule" turns up in 6 different configs.
I strongly favor this staying here. Unneeded config options should explicitly not be part of any config. Even more so, a config object should error out upon initialization if user configures something that is not part of the algorithm in the long run.

Reply:
Absolutely, that's also the solution I thought of 👍